Client/WebFilters/FilterListResourceDownloader.swift

```diff
@@ -0,0 +1,483 @@
+    
+    // Make sure we remove the old resource if there is one
+    await AdBlockEngineManager.shared.removeResources(
+      for: .adBlock,
+      resourceTypes: [.dat, .jsonResources]
+    )
+    
+    // Let's the new ones back in
+    await AdBlockEngineManager.shared.add(
+      resource: AdBlockEngineManager.Resource(type: .dat, source: .adBlock),
+      fileURL: folderURL.appendingPathComponent("rs-ABPFilterParserData.dat"),
+      version: version
+    )
+    
+    await AdBlockEngineManager.shared.add(
+      resource: AdBlockEngineManager.Resource(type: .jsonResources, source: .adBlock),
+      fileURL: folderURL.appendingPathComponent("resources.json"),
+      version: version
+    )
+  }
+  
+  /// Load filter lists from the ad block service
+  private func loadFilterLists(from regionalFilterLists: [AdblockFilterList], filterListSettings: [FilterListSetting]) -> [FilterList] {
+    return regionalFilterLists.map { adBlockFilterList in
+      let setting = filterListSettings.first(where: { $0.uuid == adBlockFilterList.uuid })
+      return FilterList(from: adBlockFilterList, isEnabled: setting?.isEnabled ?? false)
+    }
+  }
+  
+  /// Subscribe to the UI changes on the `filterLists` so that we can save settings and register or unregister the filter lists
+  private func subscribeToFilterListChanges() {
+    // Subscribe to changes on the filter list states
+    filterListSubscription = $filterLists
+      .receive(on: DispatchQueue.main)
+      .sink { filterLists in
+        for filterList in filterLists {
+          self.handleUpdate(to: filterList)
+        }
+      }
+  }
+  
+  /// Ensures settings are saved for the given filter list and that our publisher is aware of the changes
+  private func handleUpdate(to filterList: FilterList) {
+    assertIsMainThread("Not main thread")
+    
+    settingsManager.upsertSetting(
+      uuid: filterList.uuid,
+      isEnabled: filterList.isEnabled,
+      componentId: filterList.componentId
+    )
+    
+    if filterList.isEnabled {
+      register(filterList: filterList)
+    } else {
+      unregister(filterList: filterList)
+    }
+  }
+  
+  /// Register all enabled filter lists
+  private func registerAllEnabledFilterLists() {
+    for filterList in filterLists {
+      guard filterList.isEnabled else { continue }
+      register(filterList: filterList)
+    }
+  }
+  
+  /// Register this filter list and start all additional resource downloads
+  private func register(filterList: FilterList) {
+    guard adBlockServiceTasks[filterList.uuid] == nil else { return }
+    guard let adBlockService = adBlockService else { return }
+    guard let index = filterLists.firstIndex(where: { $0.uuid == filterList.uuid }) else { return }
+    
+    for resource in filterList.resources {
+      startFetching(resource: resource, for: filterList, index: index)
+    }
+
+    adBlockServiceTasks[filterList.uuid] = Task.detached(priority: .background) {
```

Comment:
Change to `Task { @MainActor in`, remove main actor jump below (no code in this closure that seems to need non-main execution, plus you don't want background priority)